islandora-support/src/main/java/ca/islandora/alpaca/support/event/AS2Object.java

```diff
@@ -0,0 +1,183 @@
+    /**
+     * @return  URN of object
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * @param   id  URN of object
+     */
+    public void setId(final String id) {
+        this.id = id;
+    }
+
+    /**
+     * @return  URLs for object
+     */
+    public AS2Url[] getUrl() {
+        return url;
+    }
+
+    /**
+     * @param   url  URLs for object
+     */
+    public void setUrl(final AS2Url[] url) {
+        this.url = url.clone();
+    }
+
+    /**
+     * @return true or false
+     */
+    @JsonProperty("isNewVersion")
+    public Boolean getIsNewVersion() {
+        return isNewVersion;
+    }
+
+    /**
+     * @param isNewVersion true or false
+     */
+    public void setIsNewVersion(final Boolean isNewVersion) {
+        this.isNewVersion = isNewVersion;
+    }
+
+    /**
+     * @return the Json-ld Url
+     * @throws MissingJsonldUrlException
+     *   When there is no url with application/ld+json mimetype
+     */
+    public AS2Url getJsonldUrl() throws MissingPropertyException {
+        return getObjectUrl("application/ld+json", null, new MissingJsonldUrlException());
+    }
+
+    /**
+     * @return the Canonical Url
+     * @throws MissingJsonUrlException
+     *   When there is no url with rel = canonical and text/html mimetype
+     */
+    public AS2Url getJsonUrl() throws MissingPropertyException {
+        return getObjectUrl("application/json", null, new MissingJsonUrlException());
+    }
+
+    /**
+     * @return the Canonical Url
+     * @throws MissingCanonicalUrlException
+     *   When there is no url with rel = canonical and text/html mimetype
+     */
+    public AS2Url getCanonicalUrl() throws MissingPropertyException {
+        return getObjectUrl(null, "canonical", new MissingCanonicalUrlException());
+    }
+
+    /**
+     * @return the Canonical Url
+     * @throws MissingDescribesUrlException
+     *   When there is no url with rel = describes
+     */
+    public AS2Url getDescribesUrl() throws MissingPropertyException {
+        return getObjectUrl(null, "describes", new MissingDescribesUrlException());
+    }
+
+    /**
+     * Utility to filter AS2Urls, filters by mimetype or rel or both
+     * @param mimetype
+     *   The mimetype to filter on or null for none.
+     * @param rel
+     *   The rel to filter on or null for none.
+     * @param e
+     *   The exception to throw if we can't find a matching url
+     * @return
+     *   The first matching AS2Url.
+     * @throws MissingPropertyException
+     *   If no matching url can be found.
+     */
+    private AS2Url getObjectUrl(final String mimetype, final String rel, final MissingPropertyException e) throws
+            MissingPropertyException {
+        if (url == null) {
+            throw e;
+        }
+        final var filterUrl = Arrays.stream(url).filter(a -> {
+            if (mimetype != null) {
+                if (a.getMediaType() == null || !a.getMediaType().equalsIgnoreCase(mimetype)) {
```

Comment:
PMD flagging here:

```
5 	/home/adam/repos/Alpaca/islandora-support/src/main/java/ca/islandora/alpaca/support/event/AS2Object.java 	169 	These nested if statements could be combined
```

Linking to https://pmd.github.io/pmd-6.21.0/pmd_rules_java_design.html#collapsibleifstatements

Reply:
Resolved in https://github.com/Islandora/Alpaca/pull/82/commits/5e5d11e3292a7c6458442aacf47b018476640f8a